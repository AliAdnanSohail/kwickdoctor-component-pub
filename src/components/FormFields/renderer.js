import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
