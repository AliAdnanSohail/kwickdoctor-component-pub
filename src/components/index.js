export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
