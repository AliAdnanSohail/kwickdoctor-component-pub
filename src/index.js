export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
