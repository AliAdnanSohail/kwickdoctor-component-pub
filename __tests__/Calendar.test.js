import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';
import { mount } from 'enzyme';

import { Calendar } from '../src';

describe('Calendar', () => {
  const testDate = '2018-03-15';
  const baseProps = {
    min: moment(testDate),
    max: moment(testDate).add(1, 'month'),
    events: [
      { date: moment(testDate), name: 'event' },
      { date: moment(testDate), name: 'event' },
      { date: moment(testDate), name: 'event' },
      { date: moment(testDate), name: 'event' },
      { date: moment(testDate), name: 'event' },
      { date: moment(testDate).add(1, 'day'), name: 'event' },
      { date: moment(testDate).add(1, 'day'), name: 'event' },
      { date: moment(testDate).add(1, 'day'), name: 'event' },
      { date: moment(testDate).add(1, 'day'), name: 'event' },
      { date: moment(testDate).add(2, 'day'), name: 'event' },
      { date: moment(testDate).add(2, 'day'), name: 'event' },
      { date: moment(testDate).add(2, 'day'), name: 'event' },
      { date: moment(testDate).add(3, 'day'), name: 'event' },
      { date: moment(testDate).add(3, 'day'), name: 'event' },
      { date: moment(testDate).add(4, 'day'), name: 'event' },
    ],
    onSelect() {},
    selected: moment(testDate),
    hasEvents: true,
  };

  it('renders correctly', () => {
    const element = renderer.create(<Calendar {...baseProps} />).toJSON();

    expect(element).toMatchSnapshot();
  });

  it('change months correctly', () => {
    const dateNext = moment(testDate).add(1, 'month');
    const datePrev = moment(testDate).add(-1, 'month');

    const element = mount(<Calendar {...baseProps} />);
    expect(element.find(`.day--date--${dateNext.format('YYYY-MM-DD')}`)).toHaveLength(0);
    expect(element.find(`.day--date--${datePrev.format('YYYY-MM-DD')}`)).toHaveLength(0);

    const nextButton = element.find('button.next');
    nextButton.simulate('click');
    expect(element.find(`.day--date--${dateNext.format('YYYY-MM-DD')}`)).toHaveLength(1);

    const prevButton = element.find('button.prev');
    prevButton.simulate('click');
    prevButton.simulate('click');
    expect(element.find(`.day--date--${datePrev.format('YYYY-MM-DD')}`)).toHaveLength(1);
  });

  describe('onSelect', () => {
    const expected = {};
    const props = {
      ...baseProps,
      onSelect: (day, events) => {
        expected.dayMoment = day;
        expected.eventsArray = events;
      },
    };

    it('should receive clicked day', () => {
      const clickDate = moment(testDate).add(1, 'day');

      const element = mount(<Calendar {...props} />);
      const button = element.find(`.day--date--${clickDate.format('YYYY-MM-DD')}`);
      button.simulate('click');

      expect(clickDate.isSame(expected.dayMoment, 'day')).toEqual(true);
    });

    it('should receive events only equal to current day', () => {
      const clickDate = moment(testDate).add(1, 'day');

      const element = mount(<Calendar {...props} />);
      const button = element.find(`.day--date--${clickDate.format('YYYY-MM-DD')}`);
      button.simulate('click');

      expect(expected.eventsArray).toHaveLength(4);
    });
  });

  describe('Dots', () => {
    it('should show 3 dots if there are more than 3 events', () => {
      const clickDate = moment(testDate).add(1, 'day');
      const element = mount(<Calendar {...baseProps} />);
      const button = element.find(`.day--date--${clickDate.format('YYYY-MM-DD')}`);
      button.simulate('click');

      expect(element.find(`.day--selected--with-events.day--date--${clickDate.format('YYYY-MM-DD')}`)).toHaveLength(1);
      expect(element.find(`.day--selected--with-events.day--date--${clickDate.format('YYYY-MM-DD')} .events__item`)).toHaveLength(3);
    });

    it('should show 3 dots if there are 3 events', () => {
      const clickDate = moment(testDate).add(2, 'day');
      const element = mount(<Calendar {...baseProps} />);
      const button = element.find(`.day--date--${clickDate.format('YYYY-MM-DD')}`);
      button.simulate('click');

      expect(element.find(`.day--selected--with-events.day--date--${clickDate.format('YYYY-MM-DD')}`)).toHaveLength(1);
      expect(element.find(`.day--selected--with-events.day--date--${clickDate.format('YYYY-MM-DD')} .events__item`)).toHaveLength(3);
    });

    it('should show 2 dots if there are 2 events', () => {
      const clickDate = moment(testDate).add(3, 'day');
      const element = mount(<Calendar {...baseProps} />);
      const button = element.find(`.day--date--${clickDate.format('YYYY-MM-DD')}`);
      button.simulate('click');

      expect(element.find(`.day--selected--with-events.day--date--${clickDate.format('YYYY-MM-DD')}`)).toHaveLength(1);
      expect(element.find(`.day--selected--with-events.day--date--${clickDate.format('YYYY-MM-DD')} .events__item`)).toHaveLength(2);
    });

    it('should show 1 dot if there is 1 event', () => {
      const clickDate = moment(testDate).add(4, 'day');
      const element = mount(<Calendar {...baseProps} />);
      const button = element.find(`.day--date--${clickDate.format('YYYY-MM-DD')}`);
      button.simulate('click');

      expect(element.find(`.day--selected--with-events.day--date--${clickDate.format('YYYY-MM-DD')}`)).toHaveLength(1);
      expect(element.find(`.day--selected--with-events.day--date--${clickDate.format('YYYY-MM-DD')} .events__item`)).toHaveLength(1);
    });

    it('does not show any dots if there are no events', () => {
      const clickDate = moment(testDate).add(5, 'day');
      const element = mount(<Calendar {...baseProps} />);
      const button = element.find(`.day--date--${clickDate.format('YYYY-MM-DD')}`);
      button.simulate('click');

      expect(element.find(`.day--selected--with-events.day--date--${clickDate.format('YYYY-MM-DD')}`)).toHaveLength(0);
      expect(element.find(`.day--selected--without-events.day--date--${clickDate.format('YYYY-MM-DD')}`)).toHaveLength(1);
    });
  });

  it('should set min and max correctly', () => {
    const props = Object.assign({}, baseProps);
    let selectCount = 0;
    props.onSelect = () => {
      selectCount += 1;
    };
    const minDate = moment(testDate);
    const disabledDate1 = moment(testDate).add(-1, 'day');
    const disabledDate2 = moment(testDate).add(1, 'month').add(1, 'day');
    const element = mount(<Calendar {...props} />);

    element.find(`.day--date--${minDate.format('YYYY-MM-DD')}`).simulate('click');
    expect(selectCount === 1).toEqual(true);
    element.find(`.day--disabled.day--date--${disabledDate1.format('YYYY-MM-DD')}`).simulate('click');
    element.find('button.next').simulate('click');
    element.find(`.day--disabled.day--date--${disabledDate2.format('YYYY-MM-DD')}`).simulate('click');
    expect(selectCount === 1).toEqual(true);
  });
});
