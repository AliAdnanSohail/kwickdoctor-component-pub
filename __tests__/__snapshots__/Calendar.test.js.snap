// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar renders correctly 1`] = `
<div
  className="jsx-1398186111 calendar"
>
  <div
    className="jsx-3590812337"
  >
    <div
      className="jsx-1167149914 navigation"
    >
      <button
        className="jsx-1167149914 prev"
        onClick={[Function]}
      >
        <i
          className="material-icons 36 md-dark "
          style={
            Object {
              "color": "#666",
              "fontSize": 36,
            }
          }
        >
          navigate_before
        </i>
      </button>
      <div
        className="jsx-1167149914 navigation__month"
      >
        March
        <span
          className="jsx-1167149914"
        >
          Â·
        </span>
        2018
      </div>
      <button
        className="jsx-1167149914 next"
        onClick={[Function]}
      >
        <i
          className="material-icons 36 md-dark "
          style={
            Object {
              "color": "#666",
              "fontSize": 36,
            }
          }
        >
          navigate_next
        </i>
      </button>
    </div>
    <ul
      className="jsx-3590812337 weekdays"
    >
      <li
        className="jsx-3590812337 weekdays__item"
      >
        Sun
      </li>
      <li
        className="jsx-3590812337 weekdays__item"
      >
        Mon
      </li>
      <li
        className="jsx-3590812337 weekdays__item"
      >
        Tue
      </li>
      <li
        className="jsx-3590812337 weekdays__item"
      >
        Wed
      </li>
      <li
        className="jsx-3590812337 weekdays__item"
      >
        Thu
      </li>
      <li
        className="jsx-3590812337 weekdays__item"
      >
        Fri
      </li>
      <li
        className="jsx-3590812337 weekdays__item"
      >
        Sat
      </li>
    </ul>
  </div>
  <ul
    className="jsx-893175612 month"
  >
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-02-25"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          25
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-02-26"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          26
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-02-27"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          27
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-02-28"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          28
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-01"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          1
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-02"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          2
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-03"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          3
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-04"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          4
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-05"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          5
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-06"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          6
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-07"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          7
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-08"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          8
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-09"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          9
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-10"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          10
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-11"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          11
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-12"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          12
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-13"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          13
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--disabled"
        data-day="2018-03-14"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          14
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day day--selected day--has-events"
        data-day="2018-03-15"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          15
        </span>
        <ul
          className="jsx-341426170 events"
        >
          <li
            className="jsx-341426170 events__item"
          />
          <li
            className="jsx-341426170 events__item"
          />
          <li
            className="jsx-341426170 events__item"
          />
        </ul>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-16"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          16
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-17"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          17
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-18"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          18
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-19"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          19
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-20"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          20
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-21"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          21
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-22"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          22
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-23"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          23
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-24"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          24
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-25"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          25
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-26"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          26
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-27"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          27
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-28"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          28
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-29"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          29
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-30"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          30
        </span>
      </button>
    </li>
    <li
      className="jsx-893175612"
    >
      <button
        className="jsx-1058092368 day"
        data-day="2018-03-31"
        onClick={[Function]}
        type="button"
      >
        <span
          className="jsx-1058092368 day__border"
        >
          31
        </span>
      </button>
    </li>
  </ul>
</div>
`;
